### ISSUE TITLE
mcdespot: noisy results

### ISSUE THREAD

**Issue by andersonwinkler** [spinicist/QUIT#29]
Hi Tobias,

Thank you again for making this fantastic resource available.

We run `qimcdespot` using an old version of QUIT, and the results were generally ok, although some voxels did have strange, noisy values; see the figure below the `3C_f_m` output, the old version is on the left.

With the newest version, with or without smoothing of B1, the result is in fact much worse, and affects all the GM. See below on the right side, also the `3C_f_m` output.

<img width="1189" alt="mcdespot_issue" src="https://user-images.githubusercontent.com/1075764/115566904-ad333b80-a288-11eb-9176-4d80d91f1a22.png">

Not sure what the problem could be. We invoke the current version like this:

```
qi mcdespot SPGR.nii.gz SSFP.nii.gz --B1=HIFI_B1_smooth.nii.gz --f0=FM_f0.nii.gz --mask=mask.nii.gz --json=quit.json --threads=16 --scale --model=3 --out=quit/
```

where the json is, as before:

```
{
    "MPRAGE": {
        "FA": 5,
        "TR": 0.005,
        "TI": 0.45,
        "TD": 0,
        "eta": 1,
        "ETL": 96,
        "k0": 0
    },
    "SPGR": {
        "TR": 0.005,
        "TE": 0.00184,
        "FA": [3, 4, 5, 6, 7, 9, 13, 18]
    },
    "SSFP": {
        "TR": 0.003256,
        "PhaseInc": [0, 0, 0, 0, 0, 0, 0, 0, 180, 180, 180, 180, 180, 180, 180, 180],
        "FA": [11.4705, 15.294, 20.0735, 25.808, 31.544, 38.235, 48.75, 65, 11.4705, 15.294, 20.0735, 25.808, 31.544, 38.235, 48.75, 65]
    }
}
```

Thanks!

All the best,

Anderson

**Comment by spinicist** [comment]
Hello,

First - are you aware of this paper: https://linkinghub.elsevier.com/retrieve/pii/S1053811919302496 ?

Given that white matter looks broadly similar, and the problems are mostly in CSF, I suspect something is wrong with the fitting bounds for the 3rd component in the model. Which kind of neatly illustrates the problem in the paper, but I digress.

Is it possible for you to put the data somewhere I can download?

**Comment by andersonwinkler** [comment]
Thanks! I've separated files for this subject (i.e., the inputs to `qi mcdespot`). I can post the link here but I'd like to encrypt it. How to make sure only you can decrypt? Perhaps you have a public key somewhere, or maybe I can send you the password via email or other means? Thanks thanks!
Anderson

**Comment by spinicist** [comment]
My work e-mail is at the top of the README 😉

**Comment by andersonwinkler** [comment]
Thank you! Just sent a link by email. Thanks!
Anderson

**Comment by spinicist** [comment]
For posterity, @andersonwinkler have discussed this issue over e-mail, and a fix will be available in v3.3, which is building now.
